### PR TITLE
Dev haokang

### PR DESCRIPTION
**Pull request by dustinke11**

### CODE REVIEW

**Comment by dustinke11** [comment]
@bittopaz

File: exporter/exporter.go [hunk 1]
```diff
@@ -95,8 +96,71 @@ func EcsClient() (client *ecs.Client) {
 	return client
 }
 
+func NewRedisClient() (client *r_kvstore.Client) {
```
Comment (bittopaz) [inline exporter/exporter.go]: Please use https://golang.org/doc/effective_go.html#embedding to do some refactoring on this part.

We can create a base struct 

```go
type Client struct {
    Config alicloud.Config
}

type EcsClient struct {
    *Client
}

type RedisClient struct {
    *Client
}
```

File: exporter/exporter.go [hunk 2]
```diff
@@ -95,8 +96,71 @@ func EcsClient() (client *ecs.Client) {
...
+	var i alicloudAccessConfig
+	var err error
+	if os.Getenv("ALICLOUD_DEFAULT_REGION") == "" ||
+		os.Getenv("ALICLOUD_ACCESS_KEY") == "" ||
+		os.Getenv("ALICLOUD_SECRET_KEY") == "" {
+		//get rolename
```
Comment (bittopaz) [inline exporter/exporter.go]: For the credential management, you can refer here https://github.com/aws/aws-sdk-go/tree/master/aws/credentials and learn from it.

File: exporter/redis.go [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+	if err != nil {
+		fmt.Println("json err", err)
+	}
+	err = ioutil.WriteFile(filePath, jsonScrapeConfig, 0644)
+	if err != nil {
+		panic(err)
+	}
+}
+
+func getEnv(input string) (output string) {
```
Comment (bittopaz) [inline exporter/redis.go]: rename it to splitEnvFromName

**Review by bittopaz — CHANGES_REQUESTED** [review]

**Pull request merged**